File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    count=0
    for n in lottos:
        if n in win_nums:
            count+=1
            
    if count==0: count+=1
    answer.append(7-count) # 최저 순위

    for n in lottos:
        if n==0: count+=1
        
    if count==7: count-=1
    answer.append(7-count) # 최고 순위
    
    return sorted(answer)

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    var answer = [];
    let count = 0;
    
    lottos.forEach((n) => {
        win_nums.forEach((m) => {
           if (n == m) {
               count += 1
           } 
        });
    });
    
    if (count == 0) count += 1;
    answer.push(7 - count);
    
    lottos.forEach((n) => {
       if (n == 0) {
           count += 1
       } 
    });
    
    if (count == 7) count -= 1;
    answer.push(7 - count)
    
    return answer.sort();
}